open_mastr/xml_download/utils_write_to_database.py

```diff
@@ -302,44 +302,51 @@ def write_single_entries_until_not_unique_comes_up(
 
 
 def add_missing_column_to_table(
-    err: Error, engine: sqlalchemy.engine.Engine, xml_tablename: str
+    err: Error,
+    engine: sqlalchemy.engine.Engine,
+    xml_tablename: str,
+    df: pd.DataFrame,
```

Comment:
Why not passing required_columns: List[str] instead of the whole dataframe?

Reply:
I think that I prefer to have this in the sub function, so that the code within the main function stays more abstract. What would be your argument for creating the List in the main function?

Reply:
My main argument is cleanness: The objective of the function is to add missing columns to the table. Therefor a list of required columns should ideally be one of the inputs (and would be what I would assume).

Second but weaker argument is safeness: There could be potential side effects on the df if one would manipulate it in the method (as it is a mutable object). Thus it is safer to just pass what is needed.

Reply:
Adapted in latest commit